$NetBSD: patch-make_lib_Awt2dLibraries.gmk,v 1.1 2019/10/03 16:57:20 tnn Exp $

bundled harfbuzz code needs to use -std=c++11 for static_assert().
But we can't add it to USE_LANGUAGES because the rest of the JDK
contains legacy code that requires -std=gnu++98.

--- make/modules/java.desktop/lib/Awt2dLibraries.gmk.orig	2021-04-23 22:52:34.000000000 -0400
+++ make/modules/java.desktop/lib/Awt2dLibraries.gmk	2021-06-10 13:15:59.080512977 -0400
@@ -568,7 +568,7 @@
         AccelGlyphCache.c, \
     TOOLCHAIN := TOOLCHAIN_LINK_CXX, \
     CFLAGS := $(CFLAGS_JDKLIB) $(LIBFONTMANAGER_CFLAGS), \
-    CXXFLAGS := $(CXXFLAGS_JDKLIB) $(LIBFONTMANAGER_CFLAGS), \
+    CXXFLAGS := $(CXXFLAGS_JDKLIB) -std=gnu++11 $(LIBFONTMANAGER_CFLAGS), \
     OPTIMIZATION := $(LIBFONTMANAGER_OPTIMIZATION), \
     CFLAGS_windows = -DCC_NOEX, \
     EXTRA_HEADER_DIRS := $(LIBFONTMANAGER_EXTRA_HEADER_DIRS), \
